apps/lib/droplet/droplet-full.js

```diff
@@ -15173,18 +15180,18 @@ exports.Parser = Parser = (function() {
           switch (mark.token.type) {
             case 'indentStart':
               if ((stack != null ? (ref7 = stack[stack.length - 1]) != null ? ref7.type : void 0 : void 0) !== 'block') {
-                throw new Error('Improper parser: indent must be inside block, but is inside ' + (stack != null ? (ref8 = stack[stack.length - 1]) != null ? ref8.type : void 0 : void 0));
+                throw new Error("Line " + mark.location.line + ". Improper parser: indent must be inside block, but is inside " + (stack != null ? (ref8 = stack[stack.length - 1]) != null ? ref8.type : void 0 : void 0));
```

Comment:
i'm still nervous about the fragile dependency between this error and [where we check for it](https://github.com/code-dot-org/code-dot-org/pull/35413/files#r444543628). maybe it's enough to add a comment here that points to the dependency in `dropletCommon.js` so that we don't accidentally change/translate this string and break that dependency

Reply:
Good call. Ideally, these would be typed errors, but usages of this blow up pretty quickly as we go up the stack. It's unclear how we or other droplet users would be affected by us changing this to an object rather than a string. One in-between option would be to throw a stringified json object rather than a plain string. Then at least we could get the benefits of an object without breaking everybody else.

Curious for you thoughts on that ^

Reply:
agreed about not changing this to a typed error. are we sure that throwing a stringified JSON object wouldn't break other consumers? maybe i'm not visualizing that solution properly. if we're confident that won't break other consumers, i like that idea. otherwise, i think it's fine to add a comment above this line that explains the dependency